### PR TITLE
fix(Scripts/LBRS): Quartermaster Zigris

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  Rescript Quartermaster Zigris
-  Casts hook net + moves away to start shooting
- Chases targets that are not in line of sight

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9033

## Tests Performed:

- Tested in-game.
- 

## How to Test the Changes:

1.Engage him & try his abilities
2.
3.

## Known Issues and TODO List:

- [ ] Potion cast missing
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
Tested

- [x] Casts hook net + moves away to start shooting
- [x] Chases targets that are not in line of sight
works as described

**Comment by Nyeriah** [comment]
Apparently he should run away when the throws bombs too (only targets main target?) and his bombs never miss.

https://user-images.githubusercontent.com/47818697/140967301-1b4305b7-0f65-4945-8263-2a269a7bafe0.mp4

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            {
+                me->GetMotionMaster()->MoveBackwards(me->GetVictim(), 10.0f);
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
```
Comment (FrancescoBorzi) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp:88]:
```suggestion
        if (!UpdateVictim())
        {
            return;
        }
```

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
```
Comment (FrancescoBorzi) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp:95]:
```suggestion
        if (me->HasUnitState(UNIT_STATE_CASTING))
        {
            return;
        }
```

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                        SetCombatMovement(true);
+                        me->GetMotionMaster()->Clear();
+                        me->GetMotionMaster()->MoveChase(me->GetVictim());
+                    }
+                    events.RepeatEvent(2000);
+                    break;
+            }
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
```
Comment (FrancescoBorzi) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp:134]:
```suggestion
            if (me->HasUnitState(UNIT_STATE_CASTING))
            {
                return;
            }
```

**Review by FrancescoBorzi — APPROVED** [review]

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    {
+        if (!UpdateVictim())
+        {
+            return;
+        }
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_quartermaster_zigris.cpp:95]:
```suggestion
        if (me->HasUnitState(UNIT_STATE_CASTING))
        {
            return;
        }
```

**Pull request merged**